operator/src/lib.rs

```diff
@@ -75,37 +76,142 @@ impl NifiState {
+                        debug!(
+                            "ConfigMap [{}] already exists with identical data, skipping creation!",
+                            cm_name
+                        );
+                        return Ok(());
+                    } else {
+                        debug!(
+                            "ConfigMap [{}] already exists, but differs, recreating it!",
+                            cm_name
+                        );
+                    }
+                }
             }
             Err(e) => {
                 // TODO: This is shit, but works for now. If there is an actual error in comes with
                 //   K8S, it will most probably also occur further down and be properly handled
-                debug!("Error getting ConfigMap [{}]: [{:?}]", name, e);
+                debug!("Error getting ConfigMap [{}]: [{:?}]", cm_name, e);
             }
         }
 
-        let config = create_config_file(config);
-
-        let mut data = BTreeMap::new();
-        data.insert("nifi.properties".to_string(), config);
-
-        // And now create the actual ConfigMap
-        let cm =
-            stackable_operator::config_map::create_config_map(&self.context.resource, &name, data)?;
+        let cm = stackable_operator::config_map::create_config_map(
+            &self.context.resource,
+            &cm_name,
+            data,
+        )?;
         self.context.client.create(&cm).await?;
         Ok(())
     }
 
+    async fn wait_for_host_name_and_create_config_map(&self) -> NifiReconcileResult {
+        for pod in &self.existing_pods {
+            let pod_name = match &pod.metadata.name {
+                Some(name) => name,
+                None => {
+                    error!("Pod has no name set. This should not happen!");
+                    return Ok(ReconcileFunctionAction::Done);
+                }
+            };
+
+            let spec = match &pod.spec {
+                Some(spec) => spec,
+                None => {
+                    debug!("Pod [{}] has no spec set yet. Waiting...!", &pod_name);
+                    return Ok(ReconcileFunctionAction::Requeue(Duration::from_secs(10)));
+                }
+            };
+
+            let node_name = match &spec.node_name {
+                Some(node_name) => node_name,
+                None => {
+                    debug!("Pod [{}] has no node_name set yet. Waiting...", &pod_name);
+                    return Ok(ReconcileFunctionAction::Requeue(Duration::from_secs(10)));
+                }
+            };
+
+            let role_group = match pod_utils::get_pod_label(pod, APP_ROLE_GROUP_LABEL) {
+                Some(role_group) => role_group,
+                None => {
+                    error!(
+                        "Pod is missing label [{}]. This is a bug and should not happen!",
+                        APP_ROLE_GROUP_LABEL
+                    );
+                    continue;
+                }
+            };
```

Comment:
I believe all of this can be done by `delete_illegal_pods` in operator-rs

Reply:
I dont understand? Im waiting here or make sure that the pod has a host/node name. It should not be deleted. The nodename is required for the config properties in the config map. 
If you create the pod and reuse the pod returned by the create, it can happen that it is not scheduled yet and does not have a nodename. 
I think this is currently only required in NiFi, the other operators a fine.